Fix SIGABRT caused by double exception in PyTorchStreamReader when file not found.

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33243 Fix SIGABRT caused by double exception in PyTorchStreamReader when file not found.**

If a file does not exist in an archive, PyTorchStreamReader throws an exception. However, when PyTorchStreamReader is destructed another exception is thrown while processing the first exception. As a result of this double exception there is SIGABRT.

Thanks @dreiss for catching this bug and suggesting the fix. It happened when he used _load_for_mobile to load a torch script file without bytecode session. A unittest is added to test this case.

Differential Revision: [D19859205](https://our.internmc.facebook.com/intern/diff/D19859205)

Commit:
Fix SIGABORT caused by double exception in PyTorchStreamReader when file not found.

dreiss reviewed (approved):
nit: It's spelled "SIGABRT", because apparently that seemed like a good idea in the 70s.

Commit:
Update on "Fix SIGABRT caused by double exception in PyTorchStreamReader when file not found."


If a file does not exist in an archive, PyTorchStreamReader throws an exception. However, when PyTorchStreamReader is destructed another exception is thrown while processing the first exception. As a result of this double exception there is SIGABRT.

Thanks @dreiss for catching this bug and suggesting the fix. It happened when he used _load_for_mobile to load a torch script file without bytecode session. A unittest is added to test this case.

Differential Revision: [D19859205](https://our.internmc.facebook.com/intern/diff/D19859205)
